### ISSUE TITLE
Postgres to Snowflake sync failed for 10M rows table

### ISSUE THREAD

**Issue by tagnev** [airbytehq/airbyte#15210]
Hi, 

I am trying to sync the data from POSTGRES to SNOWFLAKE of 10M rows table. But after pulled 600k rows got stuck on below error,
2022-08-02 03:54:44 INFO i.a.w.g.DefaultReplicationWorker(lambda$getReplicationRunnable$6):325 - Records read: 698000 (6 GB)
2022-08-02 03:54:45 destination > 2022-08-02 03:54:45 INFO i.a.i.d.r.SerializedBufferingStrategy(flushWriter):93 - Flushing buffer of stream MTL_SYSTEM_ITEMS (200 MB)
2022-08-02 03:54:45 destination > 2022-08-02 03:54:45 INFO i.a.i.d.s.StagingConsumerFactory(lambda$flushBufferFunction$3):158 - Flushing buffer for stream MTL_SYSTEM_ITEMS (200 MB) to staging
2022-08-02 03:54:45 destination > 2022-08-02 03:54:45 INFO i.a.i.d.r.BaseSerializedBuffer(flush):131 - Wrapping up compression and write GZIP trailer data.
2022-08-02 03:54:45 destination > 2022-08-02 03:54:45 INFO i.a.i.d.r.BaseSerializedBuffer(flush):138 - Finished writing data to 4802c05e-271e-4b8c-b4d2-953dda97865a16960591700195388958.csv.gz (200 MB)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.internal.SdkFilterInputStream.close(SdkFilterInputStream.java:99)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.event.ProgressInputStream.close(ProgressInputStream.java:211)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.util.IOUtils.closeQuietly(IOUtils.java:70)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutor.closeQuietlyForRuntimeExceptions(AmazonHttpClient.java:767)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:760)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:719)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:701)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:669)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:651)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:515)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4443)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4390)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.services.s3.AmazonS3Client.doUploadPart(AmazonS3Client.java:3395)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.services.s3.AmazonS3Client.uploadPart(AmazonS3Client.java:3380)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.services.s3.transfer.internal.UploadPartCallable.call(UploadPartCallable.java:33)
2022-08-02 03:57:53 destination > at net.snowflake.client.jdbc.internal.amazonaws.services.s3.transfer.internal.UploadPartCallable.call(UploadPartCallable.java:23)
2022-08-02 03:57:53 destination > at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2022-08-02 03:57:53 destination > at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2022-08-02 03:57:53 destination > at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2022-08-02 03:57:53 destination > at java.base/java.lang.Thread.run(Thread.java:833)
2022-08-02 03:57:53 destination >

Configuration below
Service: Docker
Hardware: 32 GB RAM (Local Laptop)
OS: Windows 10 Enterprise

**Comment by natalyjazzviolin** [comment]
Hey @tagnev, thanks for opening this issue.

How long had the sync been running? With such a large number of rows it could have timed out. If that's the case, there's an environmental variable you can change:
https://github.com/airbytehq/airbyte/blob/9a07136f2b7e78fd611a3ec12c8ec6db78dc4dd1/.env#L73-L77

Could you please attach your full logs? I don't see the root cause of the issue in the ones attached. Thanks!

**Comment by tagnev** [comment]
Actually its not getting timeout. Its more related to file size issue i guess. PFB the complete logs,


[logs-14.txt](https://github.com/airbytehq/airbyte/files/9253190/logs-14.txt)

**Comment by lluisgassovillarejo** [comment]
Hi,
I am getting a timeout after three days. I have updated the .env variable `MAX_SYNC_TIMEOUT_DAYS` to 7, but it still times out after 72 hours.

Attached are screenshots of the .env file I am editing and its content.

Airbyte runs in a GCP instance, and, after 72 hours, it cancels the job (attached the logs)

I tried restarting the instance from GCP (stop and start) and running docker-compose down and docker-compose up, but none worked.

Any ideas of what I am missing?

Thanks!

[logs-168.txt](https://github.com/airbytehq/airbyte/files/9478721/logs-168.txt)

![image](https://user-images.githubusercontent.com/10463486/188171737-180f9d45-f6a2-46e3-a75c-4f475da7549a.png)
![image](https://user-images.githubusercontent.com/10463486/188171876-b990ca9a-1994-4d56-b38a-9df9d093fecb.png)
